R/plot_taxa.R

```diff
@@ -422,3 +758,548 @@ format_for_comm_plots <- function(observation, id) {
+  
+  
+  # Scale font size
+  uniy <- length(unique(flat_data$taxon_name))
+  if (uniy < 30) {
+    txty <- NULL
+  } else if (uniy < 60) {
+    txty <- 6
+  } else if (uniy >= 60) {
+    txty <- 4
+  }
+  
+  
+  # make initial plot object
+  if(!is.na(color_var)){
+    p <- flat_data %>%
+      ggplot2::ggplot(
+        ggplot2::aes(
+          x = stats::reorder(.data$taxon_name, -.data$value),
+          y = .data$value,
+          color = .data[[color_var]],
+          fill = .data[[color_var]]))
+  }else{
+    p <- flat_data %>%
+      ggplot2::ggplot(
+        ggplot2::aes(
+          x = stats::reorder(.data$taxon_name, -.data$value),
+          y = .data$value))
+  }
+  
+  # add facets if provided
+  if(!is.na(facet_var)){
+    if(length(unique(flat_data$unit)) > 1){
+      p <- p + ggplot2::facet_wrap(~ as.factor(.data[[facet_var]]) +
+                                     as.factor(.data$unit),
+                                   scales = facet_scales)
+    }else{
+      p <- p + ggplot2::facet_wrap(~ as.factor(.data[[facet_var]]),
+                                   scales = facet_scales)
+    }
+  }
+  
+  # add title, labels and axis formatting
+  p <- p +
+    ggplot2::labs(title = plot_title, subtitle = plot_subtitle) +
+    ggplot2::xlab("Taxon name") +
+    ggplot2::ylab(paste0("Abundance (",paste(unique(flat_data$unit), collapse = " or "), ")")) +
+    geom_boxplot(alpha = 0.5) + # alpha = transparency
+    ggplot2::coord_flip() +
+    ggplot2::scale_x_discrete(limits = rev) +
+    ggplot2::scale_y_continuous(trans = trans) +
+    ggplot2::theme_bw() +
+    ggplot2::theme(
+      axis.text.y.left = ggplot2::element_text(size = txty),
+      axis.text.x = ggplot2::element_text(angle = 45, hjust = 1))
+  
+  # return plot
+  return(p)
+}
+
+
+
+
+
+#' Plot sites on US map
+#'
+#' @param dataset (list) An ecocomDP formatted dataset
+#' @param flat_data (tbl_df, tbl, data.frame) A flat data.frame created from an \code{ecocomDP} formatted dataset using \code{ecocomDP::flatten_data}.
+#' @param observation (tbl_df, tbl, data.frame) The observation table.
+#' @param location (tbl_df, tbl, data.frame) The location table.
+#' @param id (character) Identifier of dataset to be used in plot subtitles.
+#' @param alpha (numeric) Alpha-transparency scale of data points. Useful when many data points overlap. Allowed values are between 0 and 1, where 1 is 100\% opaque. Default is 1.
+#' @param labels (boolean) Argument to show labels of each US state. Default is TRUE.
+#' 
+#' @return (gg, ggplot) A gg, ggplot object if assigned to a variable, otherwise a plot to your active graphics device
+#' 
+#' 
+#' @export
+#' 
+#' @examples
+#' \dontrun{
+#' plot_sites(
+#'   dataset = ants_L1)
+#' 
+#' plot_sites(
+#'   flat_data = flatten_dataset(ants_L1))
+#' 
+#' plot_sites(
+#'   observation = ants_L1[[1]]$tables$observation,
+#'   location = ants_L1[[1]]$tables$location)
+#' }
+plot_sites <- function(
```

Comment:
Same comments as `plot_taxa_accum_sites()`

Reply:
this function was originally written to require multiple tables, so this guidance doesn't work here.

Reply:
Use the argument subsuming approach offered in `plot_taxa_accum_sites()`?